mage_ai/api/resources/PipelineRunResource.py

```diff
@@ -166,23 +166,35 @@ def _create_callback(resource):
 
     @safe_db_query
     def update(self, payload, **kwargs):
-        if 'retry_blocks' == payload.get('pipeline_run_action') and \
-                PipelineRun.PipelineRunStatus.COMPLETED != self.model.status:
+        if 'retry_blocks' == payload.get('pipeline_run_action'):
             self.model.refresh()
-
             pipeline = Pipeline.get(self.model.pipeline_uuid)
-
-            incomplete_block_runs = \
-                list(
-                    filter(
-                        lambda br: br.status != BlockRun.BlockRunStatus.COMPLETED,
-                        self.model.block_runs
+            block_runs_to_retry = []
+            from_block_uuid = payload.get('from_block_uuid')
+            if PipelineRun.PipelineRunStatus.COMPLETED != self.model.status:
```

Comment:
This `if block` coming first is intentional. If a block had failed and a user tries to re-run from a specific block downstream to the failed block, we need to rerun from the failed block.